tests/app/main/views/test_users.py

```diff
@@ -362,6 +362,137 @@ def test_download_csv_for_on_framework_only(self):
+                'companies_house_number': 'ABC123',
+                "published_services_count": {
+                    "digital-outcomes": 0,
+                    "digital-specialists": 0,
+                    "user-research-studios": 0,
+                    "user-research-participants": 0,
+                },
+                "contact_information": {
+                    'contact_name': 'Contact for Supplier 1',
+                    'contact_email': '1@contact.com',
+                    'contact_phone_number': '12345',
+                    'address_first_line': '7 Gem Lane',
+                    'address_city': 'Cantelot',
+                    'address_postcode': 'CN1A 1AA',
+                    'address_country': 'country:GB',
+                },
+                'variations_agreed': '',
+            }
+        ]
+        self.data_api_client.export_suppliers.return_value = {"suppliers": list_of_expected_supplier_results}
+        self.data_api_client.find_frameworks.return_value = {"frameworks": [self._valid_framework]}
+
+        self.data_api_client.get_framework.return_value = {"frameworks": self._valid_framework}
+        response = self.client.get(
+            '/admin/frameworks/{}/suppliers/download'.format(
+                self._valid_framework['slug']
+            )
+        )
+        assert response.status_code == 200
+        assert response.mimetype == 'text/csv'
+        assert (response.headers['Content-Disposition'] ==
+                'attachment;filename=suppliers-on-digital-outcomes-and-specialists-2.csv')
+
+        rows = [line.split(",") for line in response.get_data(as_text=True).splitlines()]
+
+        assert len(rows) == len(list_of_expected_supplier_results) + 1
+        expected_top_level_headings = [
+            "supplier_id",
+            "supplier_name",
+            "supplier_organisation_size",
+            "duns_number",
+            "companies_house_number",
+            "registered_name",
+            "declaration_status",
+            "application_status",
+            "application_result",
+            "framework_agreement",
+            "variations_agreed",
+        ]
+        expected_service_count_headings = [
+            "service-count-digital-outcomes",
+            "service-count-digital-specialists",
+            "service-count-user-research-studios",
+            "service-count-user-research-participants",
+        ]
+        expected_contact_info_headings = [
+            'contact_name',
+            'contact_email',
+            'contact_phone_number',
+            'address_first_line',
+            'address_city',
+            'address_postcode',
+            'address_country',
+        ]
+
+        assert rows[0] == expected_top_level_headings + expected_service_count_headings + expected_contact_info_headings
+        # All suppliers returned from the API should appear in the CSV
+
+        for index, expected_supplier in enumerate(list_of_expected_supplier_results):
```

Comment:
You can tell enumerate where to start the index with a second arg.
http://book.pythontips.com/en/latest/enumerate.html#enumerate